### ISSUE TITLE
Labels (keypoints) go outside of the frame

### ISSUE THREAD

**Issue by aarahvar2** [talmolab/sleap#896]
# Bug description

## Expected behaviour
All keypoints should remain in the frame area.

## Actual behaviour
Sometimes, by mistake, if you move the keypoint outside of the frame and unclick the mouse, keypoint remains outside of the frame and remains inaccessible.

## Your personal set up

- OS:Windows

- Version(s):10

- SLEAP installation method (listed [here](https://sleap.ai/installation.html#)):
  - [ x] [Conda from package](https://sleap.ai/installation.html#conda-package)
  - [ ] [Conda from source](https://sleap.ai/installation.html#conda-from-source)
  - [x ] [pip package](https://sleap.ai/installation.html#pip-package)
  - [ ] [M1 Macs](https://sleap.ai/installation.html#m1-macs)

<details><summary>Environment packages</summary>

```
# paste output of `pip freeze` or `conda list` here
```

</details>

<details><summary>Logs</summary>

```
# paste relevant logs here, if any
```

</details>

## Screenshots
![Screenshot 2022-08-06 132649](https://user-images.githubusercontent.com/63943838/183242292-f4e08a06-7967-4a72-9c44-951f212b0e16.jpg)

## How to reproduce

1-Right click on a keypoint
2-move it to the outside of the frame
3- unclick the mouse

**Comment by roomrys** [comment]
Hi @aarahvar2,

This is a known bug. Thanks for reporting! Closing for now and following up in #669.

Thanks,
Liezl

**Comment by talmo** [comment]
To be clear: #669 refers to when you _change frames_ while holding down a node.

I think in this case, if you moved it out of the frame by accident without changing frames, you should be able to zoom out or click and drag over to see the part that is outside the frame.

Alternatively, you can also hold Alt and click+drag a visible node to move the entire instance over, pull the one that was too far away, then Alt+drag the whole instance back into place.

Let us know if neither of those are working for you or if your issue has more to do with #669.

**Comment by aarahvar2** [comment]
Thanks Talmo. Alt+drag is a good workaround.
For the next versions, it is a good idea to keep always the x and y of the keypoints in the frame.

**Comment by roomrys** [comment]
Re-opening this issue. I misinterpreted the original problem - seems related to #669, but not exactly the same. The same fix would be helpful in both cases, as @aarahvar2 suggests: always keep x, y of keypoints within the frame.

**Comment by roomrys** [comment]
## Relevant Code
1. Mouse press event for left click on node can have a few different modifiers applied (interested in `Alt` and no modifier events)
https://github.com/talmolab/sleap/blob/219a60fad332afb047c2d000e8bf15abcd2b7674/sleap/gui/widgets/video.py#L1423-L1459

2. Update node location for non-drag (no modifiers)
https://github.com/talmolab/sleap/blob/219a60fad332afb047c2d000e8bf15abcd2b7674/sleap/gui/widgets/video.py#L1479-L1496

**Comment by roomrys** [comment]
This issue has been resolved in the new release - install SLEAP v1.2.7 [here](https://sleap.ai/installation.html#conda-package).
